src/utils/withWidth.js

```diff
@@ -35,15 +37,19 @@ const withWidth = (options = {}) => Component => {
     };
 
     componentDidMount() {
-      this.updateWidth(window.innerWidth);
+      const win = ownerWindow(ReactDOM.findDOMNode(this.mountNode));
+      this.updateWidth(win.innerWidth);
     }
 
     componentWillUnmount() {
       this.handleResize.cancel();
     }
 
+    mountNode = null;
+
     handleResize = debounce(() => {
-      this.updateWidth(window.innerWidth);
+      const win = ownerWindow(ReactDOM.findDOMNode(this.mountNode));
```

Comment:
This appears to work better. I'm still not 100% confident in its function.

findDOMNode(this.mountNode) always returns null from componentDidMount when a width or initialWidth prop isn't provided (such as in the docs), since we render null when width isn't defined yet in props or state.

Reply:
You are right. The logic is broken. I'm surprised the CI is green.

Reply:
Maybe you can leave the withWidth file unchanged so we can merge the other changes right away?

Reply:
Can do. When findDOMNode returns `null` the `ownerWindow` function just defaults to returning the global window object.